inspire_utils/name.py

```diff
@@ -76,13 +79,30 @@ def __init__(self, name, constants=None):
             constants = ParsedName.constants
 
         self._parsed_name = HumanName(name, constants=constants)
+        self._parsed_name.capitalize()
 
         # In the case of one name part, i.e. only lastname, name parser adds the name part in one of the first, middle,
         # suffix fields (depending on input). For our use-case if a user types a name, usually would be a lastname.
-        if len(name.split()) == 1:
-            self._parsed_name.last = self._parsed_name.first + self._parsed_name.middle + self._parsed_name.suffix
+        if not self._has_more_than_one_parts(name):
```

Comment:
aren't you overthinking this? if I understand correctly, the problem is that when there's only one name, `first` is populated instead of `last`. Cannot you check whether that's the case and reassign as needed? something like
```python
if self._parsed_name.first and not self._parsed_name.last:
    self._parsed_name.last = self._parsed_name.first
    self._parsed_name.first = u''
```
or am I missing something?

Reply:
It's not only the first name, if there's are two commas the name part can be recognized as suffix.